KsApi/models/RewardTests.swift

```diff
@@ -97,4 +97,51 @@ final class RewardTests: XCTestCase {
     XCTAssertEqual(.unrestricted, reward.value?.shipping.preference)
     XCTAssertEqual("Ships anywhere in the world.", reward.value?.shipping.summary)
   }
+
+  func testJsonDecoding_WithShippingType_Anywhere() {
+    let reward = Reward.decodeJSONDictionary([
+      "id": 1,
+      "description": "Some reward",
+      "minimum": 10,
+      "converted_minimum": 12,
+      "backers_count": 10,
+      "shipping_enabled": true,
+      "shipping_type": "anywhere"
+    ])
+
+    XCTAssertNotNil(reward)
+    XCTAssertEqual(reward.value?.id, 1)
+    XCTAssertEqual(reward.value?.description, "Some reward")
+    XCTAssertEqual(reward.value?.minimum, 10)
+    XCTAssertEqual(reward.value?.convertedMinimum, 12)
+    XCTAssertEqual(reward.value?.backersCount, 10)
+    XCTAssertEqual(true, reward.value?.shipping.enabled)
+    XCTAssertEqual(.anywhere, reward.value?.shipping.type)
+  }
+
+  func testJsonDecoding_WithShippingType_SingleLocation() {
+    let reward = Reward.decodeJSONDictionary([
+      "id": 1,
+      "description": "Some reward",
+      "minimum": 10,
+      "converted_minimum": 12,
+      "backers_count": 10,
+      "shipping_enabled": true,
+      "shipping_type": "single_location",
+      "shipping_single_location": [
+        "id": 123,
+        "localized_name": "United States"
+      ]
+    ])
+
+    XCTAssertNotNil(reward)
+    XCTAssertEqual(reward.value?.id, 1)
+    XCTAssertEqual(reward.value?.description, "Some reward")
+    XCTAssertEqual(reward.value?.minimum, 10)
+    XCTAssertEqual(reward.value?.convertedMinimum, 12)
+    XCTAssertEqual(reward.value?.backersCount, 10)
+    XCTAssertEqual(true, reward.value?.shipping.enabled)
+    XCTAssertEqual(.singleLocation, reward.value?.shipping.type)
+    XCTAssertEqual(.init(id: 123, localizedName: "United States"), reward.value?.shipping.location)
```

Comment:
I guess same question here about testing all the remaining cases (`noShipping` & `multipleLocations`)?

Reply:
Sure, let me add those.